packages/atomic/src/components/search/atomic-pager/atomic-pager.spec.ts

```diff
@@ -0,0 +1,497 @@
+    });
+
+    const atomicIcon = element.shadowRoot?.querySelector(
+      '[part="previous-button-icon"]'
+    );
+    expect(atomicIcon).toHaveAttribute('icon', icon);
+  });
+
+  it('should be able to render a custom icon for the next button', async () => {
+    const icon = '<svg>random-next-icon</svg>';
+    const element = await renderPager({
+      nextButtonIcon: icon,
+    });
+
+    const atomicIcon = element.shadowRoot?.querySelector(
+      '[part="next-button-icon"]'
+    );
+    expect(atomicIcon).toHaveAttribute('icon', icon);
+  });
+
+  it('should render all expected parts', async () => {
+    const element = await renderPager();
+    const parts = locators.parts(element);
+
+    await expect.element(parts.buttons!).toBeInTheDocument();
+    await expect.element(parts.pageButtons!).toBeInTheDocument();
+    await expect.element(parts.pageButton!).toBeInTheDocument();
+    await expect.element(parts.previousButton!).toBeInTheDocument();
+    await expect.element(parts.nextButton!).toBeInTheDocument();
+    await expect.element(parts.previousButtonIcon!).toBeInTheDocument();
+    await expect.element(parts.nextButtonIcon!).toBeInTheDocument();
+  });
+
+  describe('when pager state changes', () => {
+    it('should update currentPages when pager state changes', async () => {
+      const element = await renderPager({
+        pagerState: {currentPages: [6, 7, 8, 9, 10]},
+      });
+
+      expect(element.pagerState.currentPages).toEqual([6, 7, 8, 9, 10]);
+    });
+
+    it('should update hasPreviousPage when pager state changes', async () => {
+      const element = await renderPager({
+        pagerState: {hasPreviousPage: false},
+      });
+
+      expect(element.pagerState.hasPreviousPage).toBe(false);
+    });
+
+    it('should update hasNextPage when pager state changes', async () => {
+      const element = await renderPager({
+        pagerState: {hasNextPage: false},
+      });
+
+      expect(element.pagerState.hasNextPage).toBe(false);
+    });
+  });
+
+  describe('when search status state changes', () => {
+    it('should update hasError when search status changes', async () => {
+      const element = await renderPager({
+        searchStatusState: {hasError: true},
+      });
+
+      expect(element.searchStatusState.hasError).toBe(true);
+    });
+
+    it('should update hasResults when search status changes', async () => {
+      const element = await renderPager({
+        searchStatusState: {hasResults: false},
+      });
+
+      expect(element.searchStatusState.hasResults).toBe(false);
+    });
+  });
+
+  describe('#validateProps', () => {
+    it('should allow numberOfPages of 0', async () => {
+      const consoleErrorSpy = vi
+        .spyOn(console, 'error')
+        .mockImplementation(() => {});
+
+      await renderPager({numberOfPages: 0});
+
+      expect(consoleErrorSpy).not.toHaveBeenCalled();
+    });
+
+    it('should handle numberOfPages as string number', async () => {
```

Comment:
Really what we should be testing here is that we revalidate the prop when the HTML attribute is updated during the component's lifecycle.

So you should probably also check that we throw if number-of-pages is set to -1.